provider/ansible.py

```diff
@@ -154,10 +154,17 @@ def distro_install():
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
-    try:
-        path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
-        ansible_install_policy = params['ansible_install_policy']
-        return policy_map[ansible_install_policy]()
-    else:
-        return True
+    cmd_checklist = ('ansible-playbook', 'sshpass')
+    for command in cmd_checklist:
+        try:
+            path.find_command(command)
+        except path.CmdNotFoundError:
+            if command == "ansible-playbook":
+                ansible_install_policy = params['ansible_install_policy']
+                if not policy_map[ansible_install_policy]():
```

Comment:
Any specific reason to have a if condition here. Can't we use the same method like before "return policy_map[ansible_install_policy]()" and move the line 170 ie "return True" to indented to if condition. 
one if condition comparison can be saved :)

Reply:
LGTM except some cosmetic changes mentioned above.

Reply:
@harihare Thank you for your comment and sorry for delay! Yes, there is a reason for that. I need to check commands from `cmd_checklist` one by one. If I returned directly `policy_map[ansible_install_policy]()` the iteration would end after the first item from `cmd_checklist`  returns true, so the rest is not checked at all.